crates/compiler/constrain/src/module.rs

```diff
@@ -33,6 +46,63 @@ pub fn constrain_module(
     constraint
 }
 
+fn constrain_params(
+    types: &mut Types,
+    constraints: &mut Constraints,
+    home: ModuleId,
+    constraint: Constraint,
+    (pattern_var, _, loc_pattern): &(Variable, AnnotatedMark, Loc<Pattern>),
+) -> Constraint {
+    let mut env = Env {
+        home,
+        rigids: MutMap::default(),
+        resolutions_to_make: vec![],
+    };
+
+    let index = constraints.push_variable(*pattern_var);
+    let expected_params = constraints.push_pat_expected_type(PExpected::NoExpectation(index));
+
+    let mut state = PatternState::default();
+
+    let closed_con = match loc_pattern.value {
+        Pattern::RecordDestructure {
+            whole_var: _,
+            ext_var,
+            destructs: _,
+        } => {
+            // Disallow record extension for module params
+            let empty_rec = constraints.push_type(types, Types::EMPTY_RECORD);
+            constraints.store(empty_rec, ext_var, file!(), line!())
+        }
+        _ => internal_error!("Only record destructures are allowed in module params. This should've been caught earlier."),
+    };
```

Comment:
Yeah, I can see how this might be annoying if you're in the process of removing a param from a module that's used in many places. If it's ok, I'd like to merge this as is for now because I'd like to start experimenting soon, and this might be a big change. I totally agree that what you suggest sounds like the way to go when things are more stable.